crates/astria-composer/src/searcher.rs

```diff
@@ -147,9 +163,25 @@ impl Searcher {
         let rollup_chain_id = cfg.chain_id.clone();
         let (status, _) = watch::channel(Status::default());
 
+        // create signing key for sequencer txs and get initial nonce
+        let private_key_bytes: [u8; 32] = hex::decode(&cfg.private_key)
+            .wrap_err("failed to decode private key bytes from hex string")?
+            .try_into()
+            .map_err(|_| eyre!("invalid private key length; must be 32 bytes"))?;
+        let sequencer_key =
+            SigningKey::try_from(private_key_bytes).wrap_err("failed to parse sequencer key")?;
+        let address = Address::from_verification_key(&sequencer_key.verification_key());
+        let starting_nonce = sequencer_client
+            .inner
+            .get_nonce(&address, None)
```

Comment:
Please add a timeout to this feature. You can use `backon`.

Reply:
I moved this to after `wait_for_seq` in `Searcher::run()`, makes more sense to fetch nonce there